### PR TITLE
test: cypress init

### PR DESCRIPTION
**Pull request by shijiatongxue**
[中文模板 / Chinese Template](https://github.com/DouyinFE/semi-design/blob/main/.github/PULL_REQUEST_TEMPLATE.zh-CN.md)

- [x] I have read and followed [Pull Request Guidelines](https://github.com/DouyinFE/semi-design/blob/main/CONTRIBUTING-en-US.md#pull-request-guidelines) of the contributing guide.

### What kind of change does this PR introduce? (check at least one)

 - [x] CI/CD improve

### PR description

Add Cypress E2E(end-to-end) test

Changes:
- Add `cypress` dev dependency to root `package.json` and update `yarn.lock`
- Add `cypress` dir to project root directory
- Add `two` demo e2e test case to `cypress/integration` dir
- Add cypress github actions
- Give a comment in PR after running github actions

image

### Additional information

Cypress is suitable for the following things

> Supplement the areas that Jest + Enzyme can't test

- Use the browser API to test components, like `scrollTo`
- Test Submit form
- Test navigation jump
- Use image comparison to test component styles
- Multi-browser test (currently only supports Chrome, Firefox, Edge for now)

[Cypress recorder Chrome extension](https://chrome.google.com/webstore/detail/cypress-recorder/glcapdcacdfkokcmicllhcjigeodacab)
Check test result in  [Cypress dashboard](https://dashboard.cypress.io/projects/k83u7j)
[Write your first test](https://docs.cypress.io/guides/getting-started/writing-your-first-test)

### CODE REVIEW

File: test/setup.js [hunk 1]
```diff
@@ -27,6 +28,7 @@ global.shallow = shallow;
 global.render = render;
 global.mount = mount;
 global.sinon = sinon;
+global.cy = cy;
```
Comment (pointhalo) [inline test/setup.js:31]: 这个是jest消费的setup，为什么要引入cypress变量定义？会使用到吗？

Comment (shijiatongxue) [reply test/setup.js:31]: 不会用到，我删除一下。

Comment (shijiatongxue) [reply test/setup.js:31]: 这个当时是因为没有语法提示，在这里引入了之后就有语法提示了。我再看下。

Comment (shijiatongxue) [reply test/setup.js]: 根据 cypress 的文档更新了一下，https://docs.cypress.io/guides/tooling/IDE-integration#Intelligent-Code-Completion

**Comment by shijiatongxue** [comment]
[Cypress 测试范围（lark doc）](https://bytedance.feishu.cn/docx/doxcnLg7thjjVLeg3k3OXrvpe4d)

与 Jest + Enzyme 的单元测试有所区别，E2E 测试前者测试不了的内容。主要包括以下场景：
- 滚动行为：Anchor、BackTop、Table、ScrollList、DatePicker、TimePicker
- 表单提交：Form
- 延时：Notification、Toast
- 链接跳转：Navigation
- 复杂用例：Tree、TreeSelect、Table

不适合测试
- 频繁变更的示例，如文案变更、DOM 变更、属性变更

**Pull request merged**